src/timepicker/timepicker.ts

```diff
@@ -0,0 +1,187 @@
+import {Component, Input, forwardRef} from '@angular/core';
+import {ControlValueAccessor, NG_VALUE_ACCESSOR} from '@angular/common';
+
+import {toInteger} from '../util/util';
+import {NgbTime} from './ngb-time';
+
+const NGB_TIMEPICKER_VALUE_ACCESSOR = {
+  provide: NG_VALUE_ACCESSOR,
+  useExisting: forwardRef(() => NgbTimepicker),
+  multi: true
+};
+
+/**
+ * A lightweight & configurable timepicker directive.
+ */
+@Component({
+  selector: 'ngb-timepicker',
+  styles: [`
+    .chevron::before {
+      border-style: solid;
+      border-width: 0.29em 0.29em 0 0;
+      content: '';
+      display: inline-block;
+      height: 0.70em;
+      left: 0.05em;
+      position: relative;
+      top: 0.15em;
+      transform: rotate(-45deg);
+      vertical-align: middle;
+      width: 0.71em;
+    }
+    
+    .chevron.bottom:before {
+      top: -.3em;
+      transform: rotate(135deg);
+    }
+  `],
```

Comment:
need to add CSS:
```
    .btn-link {
      outline: 0;
    }
```
otherwise we get the button halo affect and in IMO looks weird because it's not really a button.